src/coprocessor/dag/dag.rs

```diff
@@ -125,10 +134,19 @@ impl DAGContext {
                         chunk.mut_rows_data().extend_from_slice(&value);
                     }
                 }
-                None => {
+                Ok(None) => {
                     finished = true;
                     break;
                 }
+                Err(Error::Eval(err)) => {
+                    let mut resp = Response::new();
+                    let mut sel_resp = StreamResponse::new();
+                    sel_resp.set_error(err);
+                    let data = box_try!(sel_resp.write_to_bytes());
+                    resp.set_data(data);
+                    return Ok((Some(resp), true));
```

Comment:
I think wether there are eval errors or not shouldn't break the loop.

Reply:
+1  previous semantic is to continue the loop.

Reply:
also please add a test for this in case of mistakes in future.

Reply:
no, according to the eval error's definition(@XuHuaiyu ), here error means unforgivable  and we should stop to process the later data,  while we continue the loop when we meet warnings.  @hicqu @breeswish

Reply:
And the corresponding tests for eval error is already in `tests/coprocessor/test_select.rs`

Reply:
Ok